pkg/services/ngalert/state/historian/loki.go

```diff
@@ -378,6 +370,14 @@ func buildLogQuery(query models.HistoryQuery) (string, error) {
 
 	logQL := selectorString(selectors)
 
+	if queryHasLogFilters(query) {
+		logQL = fmt.Sprintf("%s | json", logQL)
+	}
+
+	if query.RuleUID != "" {
+		logQL = fmt.Sprintf("%s | ruleUID=%q", logQL, query.RuleUID)
+	}
```

Comment:
FYI - When dealing with a large number of log lines, Loki is often faster at filtering first on the raw log line, rather than parsing it into json then filtering, e.g.

`{...} |= "rule_uid=\"...\"" | json | ...`

But we shouldn't assume without measuring the difference on real data, so as this is a cleaner way to do it, lets leave it as and keep this as an optimization if needed.